GitHub runners fail with linker error: "collect2: error: ld returned 1 exit status"

## Motivation

Sometimes our GitHub actions tests fail with:

> Running `rustdoc --edition=2021 --crate-type lib --crate-name zebrad
> ...
>   = note: collect2: error: ld returned 1 exit status
> ...
> src/components/mempool/crawler.rs - components::mempool::crawler (line 16)

https://github.com/ZcashFoundation/zebra/actions/runs/3411951675/jobs/5676818960

This happened in the `Test beta on ubuntu-latest --features getblocktemplate-rpcs` job in PR #5565.

### Diagnostics

This usually seems to happen on beta Rust with `--features getblocktemplate-rpcs`, and only on GitHub Actions runners.

It could be a full disk, or a Rust build bug caused by rebuilding with extra features. It might also be a bug in Zebra's feature settings in `Cargo.toml`s.

The linker failure does not have a specific error message, we need to collect more logs to diagnose it.

## Related Work

This might be fixed by #5551.

--- a/.github/workflows/continous-integration-os.yml
+++ b/.github/workflows/continous-integration-os.yml
@@ -98,6 +98,10 @@ jobs:
         #       or remove this workaround once the build is more efficient (#3005).
         #with:
         #  workspaces: ". -> C:\\zebra-target"
+        with:
+          # Split the getblocktemplate-rpcs cache from the regular cache, to avoid linker errors.
+          # (These might be "disk full" errors, or they might be dependency resolution issues.)
+          key: ${{ matrix.features }}
 
       - name: Change target output directory on Windows
         # Windows doesn't have enough space on the D: drive, so we redirect the build output to the

